File: Java/Solutions/9.1. Problems For Champions - Part II/PassionDays.java
import java.math.BigDecimal;
import java.text.DecimalFormat;
import java.util.Scanner;

public class PassionDays {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        BigDecimal shoppingMoney = new BigDecimal(scanner.nextLine());

        int purchases = 0;

        String command = scanner.nextLine();

        while (!command.equals("mall.Enter")) {
            command = scanner.nextLine();
        }

        command = scanner.nextLine();

        while (!command.equals("mall.Exit")) {
            for (char action : command.toCharArray()) {

            }

            command = scanner.nextLine();
        }

        DecimalFormat formatter = new DecimalFormat("0.00");

        if (purchases == 0) {
            System.out.printf("No purchases. Money left: %s lv.\n", formatter.format(shoppingMoney));
        } else {
            System.out.printf("%d purchases. Money left: %s lv.\n", purchases, formatter.format(shoppingMoney));
        }
    }
}

File: CSharp/Solutions/9.1. Problems For Champions - Part II/PassionDays.cs
namespace Task01
{
	using System;
	
    public class PassionDays
    {
        internal static void Main(string[] args)
        {
            decimal shoppingMoney = decimal.Parse(Console.ReadLine());
            int purchases = 0;

            string command = Console.ReadLine();
            
            while (command != "mall.Enter")
            {
                command = Console.ReadLine();
            }

            command = Console.ReadLine();

            while (command != "mall.Exit")
            {
                foreach (char action in command)
                {
                    if (action >= 'A' && action <= 'Z')
                    {
                        decimal price = action * 0.5m;
                        if (shoppingMoney < price)
                        {
                            continue;
                        }

                        shoppingMoney -= price;
                        purchases++;
                    }
                    else if (action >= 'a' && action <= 'z')
                    {
                        decimal price = action * 0.3m;
                        if (shoppingMoney < price)
                        {
                            continue;
                        }

                        shoppingMoney -= price;
                        purchases++;
                    }
                    else if (action == '%')
                    {
                        if (shoppingMoney > 0)
                        {
                            shoppingMoney /= 2;
                            purchases++;
                        }
                    }
                    else if (action == '*')
                    {
                        shoppingMoney += 10;
                    }
                    else
                    {
                        decimal price = action;
                        if (shoppingMoney < price)
                        {
                            continue;
                        }

                        shoppingMoney -= price;
                        purchases++;
                    }
                }

                command = Console.ReadLine();
            }

       
            if (purchases == 0)
            {
                Console.WriteLine("No purchases. Money left: {0:f2} lv.", shoppingMoney);
            }
            else
            {
                Console.WriteLine("{0} purchases. Money left: {1:f2} lv.", purchases, shoppingMoney);
            }
	
        }
    }
}
